money-api/src/main/java/com/comcast/money/api/SpanId.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2012-2015 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.comcast.money.api;
+
+import java.util.Random;
+import java.util.UUID;
+
+/**
+ * A unique identifier for a Span
+ */
+public class SpanId {
+
+    private static final Random rand = new Random();
+    private static final String STRING_FORMAT = "SpanId~%s~%s~%s";
+
+    private final String traceId;
+    private final Long parentId;
+    private final Long selfId;
+
+    public SpanId() {
+        this(UUID.randomUUID().toString());
+    }
+
+    public SpanId(String traceId) {
+        this(traceId, rand.nextLong());
+    }
+
+    public SpanId(String traceId, Long parentId) {
+        this(traceId, parentId, rand.nextLong());
+    }
+
+    public SpanId(String traceId, Long parentId, Long selfId) {
+        this.traceId = traceId;
+        this.parentId = parentId;
+        this.selfId = selfId;
+    }
+
+    public String traceId() {
+        return traceId;
+    }
+
+    public Long parentId() {
+        return parentId;
+    }
+
+    public Long selfId() {
+        return selfId;
+    }
+
+    public SpanId newChild() {
+        return new SpanId(traceId, selfId);
+    }
+
+    @Override
+    public String toString() {
+        return String.format(STRING_FORMAT, traceId, parentId, selfId);
+    }
+
+    public static SpanId fromString(String spanIdString) {
+
+        String[] parts = spanIdString.split(",");
+        if (parts.length < 4) {
+            return null;
+        }
+
+        String traceId = parts[1].trim();
+        Long parentId = Long.parseLong(parts[2].trim());
+        Long selfId = Long.parseLong(parts[3].trim());
+        return new SpanId(traceId, parentId, selfId);
```

Comment:
If it is possible for the parent/self IDs to be null, then this probably ought to gracefully handle that. Seems like they ought to be able to round-trip.

Would also be fine if the above were primitives and defaulted to `-1` or something. Dunno.